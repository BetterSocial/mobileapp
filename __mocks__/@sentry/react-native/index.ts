export const addBreadcrumb = jest.fn();
