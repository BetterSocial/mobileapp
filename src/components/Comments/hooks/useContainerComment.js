const useContainerComment = () => {
  const isLast = (index, item, comments) =>
    index === comments.length - 1 && (item.children_counts.comment || 0) === 0;

  const isLastInParent = (index, comments) => index === comments.length - 1;

  const hideLeftConnector = (index, comments) => index === comments.length - 1;

  return {
    isLast,
    isLastInParent,
    hideLeftConnector
  };
};

export default useContainerComment;
