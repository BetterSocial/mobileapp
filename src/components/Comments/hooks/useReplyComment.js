

const useReplyComment = () => {

    const isLast = (item, index, countComment) => (
      index === countComment - 1 && (item.children_counts.comment || 0) === 0
    );

    const isLastInParent = (index, countComment) => index === countComment - 1;

  return {
    isLast,
    isLastInParent
  }

}

export default useReplyComment