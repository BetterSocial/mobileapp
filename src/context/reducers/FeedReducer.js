/* eslint-disable no-case-declarations */
import {
  SET_FEED,
  SET_FEED_BY_INDEX,
  SET_FEED_ON_SCREEN_FEED_INDEX,
  SET_FEED_TIMER,
  SET_TOPIC_FEED,
  SET_TOPIC_FEED_BY_INDEX
} from '../Types';

const feedsState = {
  feeds: [],
  topicFeeds: [],
  timer: new Date(),
  viewPostTimeIndex: 0
};
const feedsReducer = (state = feedsState, action) => {
  switch (action.type) {
    case SET_FEED:
      return {
        ...state,
        feeds: action.payload
      };

    case SET_FEED_BY_INDEX:
      const newFeeds = [...state.feeds];
      newFeeds[action.payload.index] = action.payload.singleFeed;

      return {
        ...state,
        feeds: newFeeds
      };

    case SET_TOPIC_FEED:
      return {
        ...state,
        topicFeeds: action.payload
      };

    case SET_TOPIC_FEED_BY_INDEX:
      const newTopicFeeds = [...state.topicFeeds];
      newTopicFeeds[action.payload.index] = action.payload.singleFeed;
      return {
        ...state,
        topicFeeds: newTopicFeeds
      };

    case SET_FEED_TIMER:
      return {
        ...state,
        timer: action.payload
      };

    case SET_FEED_ON_SCREEN_FEED_INDEX:
      return {
        ...state,
        viewPostTimeIndex: action.payload
      };

    default:
      return state;
  }
};
export {feedsReducer, feedsState};
