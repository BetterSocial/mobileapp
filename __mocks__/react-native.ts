import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN