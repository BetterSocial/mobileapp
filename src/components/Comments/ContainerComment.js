import * as React from 'react';
import SimpleToast from 'react-native-simple-toast';
import {Alert, Dimensions, StyleSheet, Text, TouchableWithoutFeedback, View} from 'react-native';
import {useNavigation} from '@react-navigation/native';

import ButtonHightlight from '../ButtonHighlight';
import Comment from './Comment';
import ConnectorWrapper from './ConnectorWrapper';
import LoadingComment from '../LoadingComment';
import StringConstant from '../../utils/string/StringConstant';
import useContainerComment from './hooks/useContainerComment';
import useReplyComment from './hooks/useReplyComment';
import usePostContextHook, {CONTEXT_SOURCE} from '../../hooks/usePostContextHooks';
import {colors} from '../../utils/colors';
import {deleteComment} from '../../service/comment';
import {getUserId} from '../../utils/users';
import usePostDetail from '../PostPageDetail/hooks/usePostDetail';

const ContainerComment = ({
  feedId,
  comments,
  indexFeed,
  isLoading,
  refreshComment,
  navigateToReplyView,
  findCommentAndUpdate,
  updateParentPost = () => {},
  contextSource = CONTEXT_SOURCE.FEEDS,
  itemParent
}) => {
  const navigation = useNavigation();
  const [, setSelectedCommentForDelete] = React.useState(null);
  const [selectedCommentLevelForDelete, setSelectedCommentLevelForDelete] = React.useState(0);
  const {isLast, isLastInParent, hideLeftConnector} = useContainerComment();
  const {calculationText, calculatedSizeScreen} = usePostDetail();
  const {deleteCommentFromContext} = usePostContextHook(contextSource);
  const onCommentLongPressed = async (item, level = 0) => {
    console.log('jalan2', item)
    const selfId = await getUserId();
    if (selfId === item?.user_id) {
      setSelectedCommentForDelete(item);
      setSelectedCommentLevelForDelete(level);
      Alert.alert('', StringConstant.feedDeleteCommentConfirmation, [
        {
          text: 'No, cancel',
          style: 'cancel'
        },
        {
          text: 'Yes',
          style: 'destructive',
          onPress: () => onDeleteCommentClicked(item)
        }
      ]);
    }
  };

  const onDeleteCommentClicked = async (item) => {
    const response = await deleteComment(item?.id);
    if (response?.success) {
      deleteCommentFromContext(feedId, item?.id, selectedCommentLevelForDelete, updateParentPost);
      refreshComment();
      SimpleToast.show('Comment has been deleted successfully');
    }
  };

  return (
    <View style={[styles.container]}>
      <View
        style={{
          minHeight:
            Dimensions.get('window').height -
            calculatedSizeScreen -
            calculationText(itemParent?.message, itemParent?.post_type, itemParent?.images_url)
              .containerHeight,
          borderLeftWidth: 1,
          borderLeftColor: '#C4C4C4',
          marginTop: 2
        }}>
        <View style={styles.lineBeforeProfile} />

        {comments.map((item, index) => (
          <TouchableWithoutFeedback key={index} onLongPress={() => onCommentLongPressed(item, 0)}>
            <View>
              <View key={`p${index}`}>
                {item.user ? (
                  <Comment
                    indexFeed={indexFeed}
                    key={`p${index}`}
                    comment={item}
                    user={item.user}
                    level={0}
                    time={item.created_at}
                    photo={item.user.data && item.user.data.profile_pic_url}
                    isLast={isLast(index, item, comments)}
                    isLastInParent={isLastInParent(index, comments)}
                    showLeftConnector={false}
                    onPress={() =>
                      navigateToReplyView({
                        item,
                        level: 0,
                        indexFeed
                      })
                    }
                    // refreshComment={refreshComment}
                    findCommentAndUpdate={findCommentAndUpdate}
                  />
                ) : null}
              </View>
              {item?.children_counts?.comment > 0 && (
                <ReplyComment
                  hideLeftConnector={hideLeftConnector(index, item, comments)}
                  data={item.latest_children.comment}
                  countComment={item.children_counts.comment}
                  navigation={navigation}
                  indexFeed={indexFeed}
                  navigateToReplyView={navigateToReplyView}
                  // refreshComment={(children) => refreshChildComment({parent: item, children: children.data})}
                  findCommentAndUpdate={findCommentAndUpdate}
                  onCommentLongPressed={onCommentLongPressed}
                />
              )}
            </View>
          </TouchableWithoutFeedback>
        ))}
      </View>

      {isLoading ? <LoadingComment /> : null}
    </View>
  );
};

export const ReplyComment = ({
  indexFeed,
  data,
  countComment,
  hideLeftConnector,
  navigateToReplyView,
  findCommentAndUpdate,
  onCommentLongPressed = () => {}
}) => {
  const {isLast, isLastInParent} = useReplyComment();
  return (
    <ContainerReply hideLeftConnector={hideLeftConnector}>
      {data.map((item, index) => (
        <React.Fragment key={`c-${index}`}>
          {item.user ? (
            <ConnectorWrapper index={index}>
              <TouchableWithoutFeedback onLongPress={() => onCommentLongPressed(item, 1)}>
                <View key={`c${index}`} style={styles.levelOneCommentWrapper}>
                  <Comment
                    indexFeed={indexFeed}
                    key={`c${index}`}
                    comment={item}
                    // username={item.user.data.username}
                    user={item.user}
                    level={1}
                    photo={item.user?.data?.profile_pic_url}
                    time={item.created_at}
                    onPress={() => navigateToReplyView({item, level: 2, indexFeed})}
                    isLast={isLast(item, index, countComment)}
                    // refreshComment={refreshComment}
                    findCommentAndUpdate={findCommentAndUpdate}
                  />
                  {item.children_counts.comment > 0 && (
                    <>
                      <View style={styles.seeRepliesContainer(isLastInParent(index, countComment))}>
                        <View style={styles.connector} />
                        <ButtonHightlight
                          onPress={() => navigateToReplyView({item, level: 2, indexFeed})}>
                          <Text style={styles.seeRepliesText}>
                            {StringConstant.postDetailPageSeeReplies(
                              item.children_counts.comment || 0
                            )}
                          </Text>
                        </ButtonHightlight>
                      </View>
                    </>
                  )}
                </View>
              </TouchableWithoutFeedback>
            </ConnectorWrapper>
          ) : null}
        </React.Fragment>
      ))}
    </ContainerReply>
  );
};
export const ContainerReply = ({children, isGrandchild}) => (
  <View style={[styles.containerReply, {borderColor: isGrandchild ? '#fff' : colors.gray1}]}>
    {children}
  </View>
);

export const isEqual = (prevProps, nextProps) => prevProps.comments === nextProps.comments;

export default React.memo(ContainerComment, isEqual);

export const styles = StyleSheet.create({
  container: {
    paddingLeft: 30,
    paddingRight: 8
  },
  lineBeforeProfile: {
    height: 8.5
  },
  containerReply: {
    // borderLeftWidth: 1
  },
  seeRepliesContainer: (isLast) => ({
    display: 'flex',
    flexDirection: 'row',
    paddingBottom: 14,
    borderLeftColor: isLast ? 'transparent' : colors.gray1,
    borderLeftWidth: 1
  }),
  seeRepliesText: {
    color: colors.blue
  },
  connector: {
    width: 10,
    height: 5,
    borderLeftWidth: 1.5,
    borderBottomWidth: 1.5,
    // borderBottomLeftRadius: 1,
    borderLeftColor: colors.gray1,
    borderBottomColor: colors.gray1,
    marginRight: 4,
    marginLeft: -1,
    borderBottomLeftRadius: 15 / 2,
    marginTop: 0
    // marginTop: 1
  },
  levelOneCommentWrapper: {
    flex: 1,
    marginLeft: 0
  }
});
