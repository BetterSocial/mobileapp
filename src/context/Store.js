import * as React from 'react';

import {channelReducer, channelState} from './reducers/channelReducer';
import {clientReducer, clientState} from './reducers/clientReducer';
import {discoveryReducer, discoveryState} from './reducers/discoveryReducer';
import {domainReducer, domainState} from './reducers/domainReducer';
import {feedsReducer, feedsState} from './reducers/FeedReducer';
import {followingReducer, followingState} from './reducers/followingReducer';
import {generalComponentReducer, generalComponentState} from './reducers/generalComponentReducer';
import {groupChatReducer, groupChatState} from './reducers/groupChat';
import {initialStartupReducer, initialStartupState} from './reducers/intialStartupReducer';
import {localCommunityReducer, localCommunityState} from './reducers/localComunityReducer';
import {myProfileFeedReducer, myProfileFeedState} from './reducers/myProfileFeedReducer';
import {myProfileReducer, myProfileState} from './reducers/myProfileReducer';
import {newsReducer, newsState} from './reducers/newsReducer';
import {otherProfileFeedReducer, otherProfileFeedState} from './reducers/OtherProfileFeedReducer';
import {topicsReducer, topicsState} from './reducers/topicsReducer';
import {unReadMessageReducer, unReadMessageState} from './reducers/unReadMessageReducer';
import {usersReducer, usersState} from './reducers/userReducer';
import {commentReducer, commentState} from './reducers/commentReducer';

/**
 *
 * @typedef {Object} MainContext
 * @property {import('./reducers/discoveryReducer').DiscoveryState} discovery
 */
const Store = ({children}) => {
  const rootReducer = {
    channel: React.useReducer(channelReducer, channelState),
    client: React.useReducer(clientReducer, clientState),
    discovery: React.useReducer(discoveryReducer, discoveryState),
    feeds: React.useReducer(feedsReducer, feedsState),
    following: React.useReducer(followingReducer, followingState),
    generalComponent: React.useReducer(generalComponentReducer, generalComponentState),
    groupChat: React.useReducer(groupChatReducer, groupChatState),
    localCommunity: React.useReducer(localCommunityReducer, localCommunityState),
    myProfileFeed: React.useReducer(myProfileFeedReducer, myProfileFeedState),
    otherProfileFeed: React.useReducer(otherProfileFeedReducer, otherProfileFeedState),
    news: React.useReducer(newsReducer, newsState),
    profile: React.useReducer(myProfileReducer, myProfileState),
    topics: React.useReducer(topicsReducer, topicsState),
    unReadMessage: React.useReducer(unReadMessageReducer, unReadMessageState),
    users: React.useReducer(usersReducer, usersState),
    domains: React.useReducer(domainReducer, domainState),
    initialStartup: React.useReducer(initialStartupReducer, initialStartupState),
    comments: React.useReducer(commentReducer, commentState)
  };
  return <Context.Provider value={rootReducer}>{children}</Context.Provider>;
};
export const Context = React.createContext(null);
export const useAppContext = () => React.useContext(Context);

export default Store;
