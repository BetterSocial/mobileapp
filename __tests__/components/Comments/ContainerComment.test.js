import React from 'react';
import {View} from 'react-native';
import {cleanup, fireEvent, render} from '@testing-library/react-native';
import {renderHook} from '@testing-library/react-hooks';

import useContainerComment from '../../../src/components/Comments/hooks/useContainerComment';
import useReplyComment from '../../../src/components/Comments/hooks/useReplyComment';
import ContainerComment, {
  ContainerReply,
  ReplyComment,
  isEqual
} from '../../../src/components/Comments/ContainerComment';
import {Context} from '../../../src/context/Store';
import {feedsState} from '../../../src/context/reducers/FeedReducer';

jest.mock('react-native/Libraries/Pressability/usePressability');
jest.mock('react-native/Libraries/Components/Pressable/Pressable');

jest.mock('react-native-safe-area-context', () => {
  const inset = {top: 0, right: 0, bottom: 0, left: 0};
  return {
    SafeAreaProvider: jest.fn().mockImplementation(({children}) => children),
    SafeAreaConsumer: jest.fn().mockImplementation(({children}) => children(inset)),
    useSafeAreaInsets: jest.fn().mockImplementation(() => inset)
  };
});

describe('ContainerComment should run correctly', () => {
  const comments = [
    {
      activity_id: 'eeb07336-8fe8-11ed-b458-0e0d34fb440f',
      children_counts: {},
      data: {
        count_downvote: 0,
        count_upvote: 0,
        isNotSeen: true,
        text: 'from postman 1'
      },
      id: 'dd8e0ece-a27b-4356-9a37-0427afa86258',
      kind: 'comment',
      latest_children: {},
      parent: '',
      target_feeds: [
        'notification:f19ce509-e8ae-405f-91cf-ed19ce1ed96e',
        'notification:c6c91b04-795c-404e-b012-ea28813a2006'
      ],
      updated_at: '2023-01-09T06:57:52.892334Z',
      user: {
        created_at: '2022-06-10T13:11:53.385703Z',
        data: {
          human_id: 'I4K3M10FGR78EWQQDNQ2',
          profile_pic_url:
            'https://res.cloudinary.com/hpjivutj2/image/upload/v1659099243/pbdv3jlyd4mhmtis6kqx.jpg',
          username: 'Agita'
        },
        id: 'c6c91b04-795c-404e-b012-ea28813a2006',
        updated_at: '2022-07-29T12:54:03.879150Z'
      },
      user_id: 'c6c91b04-795c-404e-b012-ea28813a2006'
    }
  ];

  const comments2 = [
    {
      activity_id: 'eeb07336-8fe8-11ed-b458-0e0d34fb440f',
      children_counts: {
        comment: 1
      },
      data: {
        count_downvote: 0,
        count_upvote: 0,
        isNotSeen: true,
        text: 'from postman 1'
      },
      id: 'dd8e0ece-a27b-4356-9a37-0427afa86258',
      kind: 'comment',
      latest_children: {
        comment: [
          {
            activity_id: 'cdfa2a76-8cb5-11ed-ab69-0e4b8d0e7a11',
            children_counts: {},
            created_at: '2023-01-09T01:37:20.225354Z',
            data: {
              count_downvote: 1,
              count_upvote: 0,
              targetFeeds: ['notification:c6c91b04-795c-404e-b012-ea28813a2006'],
              text: 'Wow'
            },
            id: '8729cdfc-ae7b-458e-a999-af967124c171',
            kind: 'comment',
            latest_children: {},
            parent: 'd9c0baea-adf7-4624-ae04-e257bd80c101',
            target_feeds: ['notification:c6c91b04-795c-404e-b012-ea28813a2006'],
            updated_at: '2023-01-09T01:38:14.677925Z',
            user: {
              created_at: '2022-06-10T13:11:53.385703Z',
              data: {
                human_id: 'I4K3M10FGR78EWQQDNQ2',
                profile_pic_url:
                  'https://res.cloudinary.com/hpjivutj2/image/upload/v1659099243/pbdv3jlyd4mhmtis6kqx.jpg',
                username: 'Agita'
              },
              id: 'c6c91b04-795c-404e-b012-ea28813a2006',
              updated_at: '2022-07-29T12:54:03.879150Z'
            },
            user_id: 'c6c91b04-795c-404e-b012-ea28813a2006'
          }
        ]
      },
      parent: '',
      target_feeds: [
        'notification:f19ce509-e8ae-405f-91cf-ed19ce1ed96e',
        'notification:c6c91b04-795c-404e-b012-ea28813a2006'
      ],
      updated_at: '2023-01-09T06:57:52.892334Z',
      user: {
        created_at: '2022-06-10T13:11:53.385703Z',
        data: {
          human_id: 'I4K3M10FGR78EWQQDNQ2',
          profile_pic_url:
            'https://res.cloudinary.com/hpjivutj2/image/upload/v1659099243/pbdv3jlyd4mhmtis6kqx.jpg',
          username: 'Agita'
        },
        id: 'c6c91b04-795c-404e-b012-ea28813a2006',
        updated_at: '2022-07-29T12:54:03.879150Z'
      },
      user_id: 'c6c91b04-795c-404e-b012-ea28813a2006'
    }
  ];

  afterEach(cleanup);
  it('Should match snapshot', () => {
    const refreshComment = jest.fn();
    const refreshChildComment = jest.fn();
    const navigateToReplyView = jest.fn();
    const findCommentAndUpdate = jest.fn();

    const contextValue = {
      feeds: [feedsState, () => jest.fn()],
      profile: [{}, jest.fn()]
    };

    const {toJSON} = render(
      <Context.Provider value={contextValue}>
        <ContainerComment
          refreshChildComment={refreshChildComment}
          navigateToReplyView={navigateToReplyView}
          findCommentAndUpdate={findCommentAndUpdate}
          refreshComment={refreshComment}
          comments={comments}
          indexFeed={undefined}
          isLoading={false}
        />
      </Context.Provider>
    );
    expect(toJSON).toMatchSnapshot();
  });

  it('ContainerReply should match snapshot', () => {
    const {toJSON, getAllByTestId} = render(
      <ContainerReply>
        <View testID="children" />{' '}
      </ContainerReply>
    );
    expect(toJSON).toMatchSnapshot();
    expect(getAllByTestId('children')).toHaveLength(1);
  });

  it('isLast level should correct', () => {
    const refreshComment = jest.fn();
    const refreshChildComment = jest.fn();
    const navigateToReplyView = jest.fn();
    const findCommentAndUpdate = jest.fn();

    const contextValue = {
      feeds: [feedsState, () => jest.fn()],
      profile: [{}, jest.fn()]
    };

    const {getAllByTestId} = render(
      <Context.Provider value={contextValue}>
        <ContainerComment
          refreshChildComment={refreshChildComment}
          navigateToReplyView={navigateToReplyView}
          findCommentAndUpdate={findCommentAndUpdate}
          refreshComment={refreshComment}
          comments={comments}
          indexFeed={undefined}
          isLoading={false}
        />
      </Context.Provider>
    );
    expect(getAllByTestId('memoComment')).toHaveLength(1);
  });

  it('react memo should change if comment has change', () => {
    expect(isEqual('1', '1')).toBeTruthy();
  });

  it('ReplyComment should match snapshot', () => {
    const navigateToReplyView = jest.fn();
    const indexFeed = 0;
    const findCommentAndUpdate = jest.fn();

    const contextValue = {
      feeds: [feedsState, () => jest.fn()],
      profile: [{}, jest.fn()]
    };

    const {toJSON} = render(
      <Context.Provider value={contextValue}>
        <ReplyComment
          countComment={comments2[0].children_counts.comment}
          indexFeed={indexFeed}
          findCommentAndUpdate={findCommentAndUpdate}
          navigateToReplyView={navigateToReplyView}
          data={comments2[0].latest_children.comment}
        />
      </Context.Provider>
    );
    expect(toJSON).toMatchSnapshot();
  });
});

describe('useContainerHooks should run correctly', () => {
  const comments2 = [
    {
      activity_id: 'eeb07336-8fe8-11ed-b458-0e0d34fb440f',
      children_counts: {
        comment: 1
      },
      data: {
        count_downvote: 0,
        count_upvote: 0,
        isNotSeen: true,
        text: 'from postman 1'
      },
      id: 'dd8e0ece-a27b-4356-9a37-0427afa86258',
      kind: 'comment',
      latest_children: {
        comment: [
          {
            activity_id: 'cdfa2a76-8cb5-11ed-ab69-0e4b8d0e7a11',
            children_counts: {},
            created_at: '2023-01-09T01:37:20.225354Z',
            data: {
              count_downvote: 1,
              count_upvote: 0,
              targetFeeds: ['notification:c6c91b04-795c-404e-b012-ea28813a2006'],
              text: 'Wow'
            },
            id: '8729cdfc-ae7b-458e-a999-af967124c171',
            kind: 'comment',
            latest_children: {},
            parent: 'd9c0baea-adf7-4624-ae04-e257bd80c101',
            target_feeds: ['notification:c6c91b04-795c-404e-b012-ea28813a2006'],
            updated_at: '2023-01-09T01:38:14.677925Z',
            user: {
              created_at: '2022-06-10T13:11:53.385703Z',
              data: {
                human_id: 'I4K3M10FGR78EWQQDNQ2',
                profile_pic_url:
                  'https://res.cloudinary.com/hpjivutj2/image/upload/v1659099243/pbdv3jlyd4mhmtis6kqx.jpg',
                username: 'Agita'
              },
              id: 'c6c91b04-795c-404e-b012-ea28813a2006',
              updated_at: '2022-07-29T12:54:03.879150Z'
            },
            user_id: 'c6c91b04-795c-404e-b012-ea28813a2006'
          }
        ]
      },
      parent: '',
      target_feeds: [
        'notification:f19ce509-e8ae-405f-91cf-ed19ce1ed96e',
        'notification:c6c91b04-795c-404e-b012-ea28813a2006'
      ],
      updated_at: '2023-01-09T06:57:52.892334Z',
      user: {
        created_at: '2022-06-10T13:11:53.385703Z',
        data: {
          human_id: 'I4K3M10FGR78EWQQDNQ2',
          profile_pic_url:
            'https://res.cloudinary.com/hpjivutj2/image/upload/v1659099243/pbdv3jlyd4mhmtis6kqx.jpg',
          username: 'Agita'
        },
        id: 'c6c91b04-795c-404e-b012-ea28813a2006',
        updated_at: '2022-07-29T12:54:03.879150Z'
      },
      user_id: 'c6c91b04-795c-404e-b012-ea28813a2006'
    }
  ];

  const twoComments = [
    {
      activity_id: 'cdfa2a76-8cb5-11ed-ab69-0e4b8d0e7a11',
      children_counts: {},
      created_at: '2023-01-09T01:37:20.225354Z',
      data: {
        count_downvote: 1,
        count_upvote: 0,
        targetFeeds: ['notification:c6c91b04-795c-404e-b012-ea28813a2006'],
        text: 'Wow'
      },
      id: '8729cdfc-ae7b-458e-a999-af967124c171',
      kind: 'comment',
      latest_children: {},
      parent: 'd9c0baea-adf7-4624-ae04-e257bd80c101',
      target_feeds: ['notification:c6c91b04-795c-404e-b012-ea28813a2006'],
      updated_at: '2023-01-09T01:38:14.677925Z',
      user: {
        created_at: '2022-06-10T13:11:53.385703Z',
        data: {
          human_id: 'I4K3M10FGR78EWQQDNQ2',
          profile_pic_url:
            'https://res.cloudinary.com/hpjivutj2/image/upload/v1659099243/pbdv3jlyd4mhmtis6kqx.jpg',
          username: 'Agita'
        },
        id: 'c6c91b04-795c-404e-b012-ea28813a2006',
        updated_at: '2022-07-29T12:54:03.879150Z'
      },
      user_id: 'c6c91b04-795c-404e-b012-ea28813a2006'
    },
    {
      activity_id: 'cdfa2a76-8cb5-11ed-ab69-0e4b8d0e7a12',
      children_counts: {},
      created_at: '2023-01-09T01:37:20.225354Z',
      data: {
        count_downvote: 1,
        count_upvote: 0,
        targetFeeds: ['notification:c6c91b04-795c-404e-b012-ea28813a2006'],
        text: 'Wow'
      },
      id: '8729cdfc-ae7b-458e-a999-af967124c171',
      kind: 'comment',
      latest_children: {},
      parent: 'd9c0baea-adf7-4624-ae04-e257bd80c101',
      target_feeds: ['notification:c6c91b04-795c-404e-b012-ea28813a2006'],
      updated_at: '2023-01-09T01:38:14.677925Z',
      user: {
        created_at: '2022-06-10T13:11:53.385703Z',
        data: {
          human_id: 'I4K3M10FGR78EWQQDNQ2',
          profile_pic_url:
            'https://res.cloudinary.com/hpjivutj2/image/upload/v1659099243/pbdv3jlyd4mhmtis6kqx.jpg',
          username: 'Agita'
        },
        id: 'c6c91b04-795c-404e-b012-ea28813a2006',
        updated_at: '2022-07-29T12:54:03.879150Z'
      },
      user_id: 'c6c91b04-795c-404e-b012-ea28813a2006'
    }
  ];

  it('isLastInParent function should run correctly', () => {
    const {result} = renderHook(() => useContainerComment());

    expect(result.current.isLastInParent(0, comments2[0].latest_children.comment)).toBeTruthy();
    expect(result.current.isLastInParent(0, twoComments)).toBeFalsy();
  });

  it('hideLeftConnector function should run correctly', () => {
    const {result} = renderHook(() => useContainerComment());
    expect(result.current.hideLeftConnector(0, comments2[0].latest_children.comment)).toBeTruthy();
    expect(result.current.hideLeftConnector(0, twoComments)).toBeFalsy();
  });

  it('isLast should run correctly', () => {
    const {result} = renderHook(() => useContainerComment());
    expect(result.current.isLast(0, comments2[0], twoComments)).toBeFalsy();
  });
});

describe('useReplyComment', () => {
  const comments2 = [
    {
      activity_id: 'eeb07336-8fe8-11ed-b458-0e0d34fb440f',
      children_counts: {
        comment: 1
      },
      data: {
        count_downvote: 0,
        count_upvote: 0,
        isNotSeen: true,
        text: 'from postman 1'
      },
      id: 'dd8e0ece-a27b-4356-9a37-0427afa86258',
      kind: 'comment',
      latest_children: {
        comment: [
          {
            activity_id: 'cdfa2a76-8cb5-11ed-ab69-0e4b8d0e7a11',
            children_counts: {
              comment: 0
            },
            created_at: '2023-01-09T01:37:20.225354Z',
            data: {
              count_downvote: 1,
              count_upvote: 0,
              targetFeeds: ['notification:c6c91b04-795c-404e-b012-ea28813a2006'],
              text: 'Wow'
            },
            id: '8729cdfc-ae7b-458e-a999-af967124c171',
            kind: 'comment',
            latest_children: {},
            parent: 'd9c0baea-adf7-4624-ae04-e257bd80c101',
            target_feeds: ['notification:c6c91b04-795c-404e-b012-ea28813a2006'],
            updated_at: '2023-01-09T01:38:14.677925Z',
            user: {
              created_at: '2022-06-10T13:11:53.385703Z',
              data: {
                human_id: 'I4K3M10FGR78EWQQDNQ2',
                profile_pic_url:
                  'https://res.cloudinary.com/hpjivutj2/image/upload/v1659099243/pbdv3jlyd4mhmtis6kqx.jpg',
                username: 'Agita'
              },
              id: 'c6c91b04-795c-404e-b012-ea28813a2006',
              updated_at: '2022-07-29T12:54:03.879150Z'
            },
            user_id: 'c6c91b04-795c-404e-b012-ea28813a2006'
          }
        ]
      },
      parent: '',
      target_feeds: [
        'notification:f19ce509-e8ae-405f-91cf-ed19ce1ed96e',
        'notification:c6c91b04-795c-404e-b012-ea28813a2006'
      ],
      updated_at: '2023-01-09T06:57:52.892334Z',
      user: {
        created_at: '2022-06-10T13:11:53.385703Z',
        data: {
          human_id: 'I4K3M10FGR78EWQQDNQ2',
          profile_pic_url:
            'https://res.cloudinary.com/hpjivutj2/image/upload/v1659099243/pbdv3jlyd4mhmtis6kqx.jpg',
          username: 'Agita'
        },
        id: 'c6c91b04-795c-404e-b012-ea28813a2006',
        updated_at: '2022-07-29T12:54:03.879150Z'
      },
      user_id: 'c6c91b04-795c-404e-b012-ea28813a2006'
    }
  ];

  it('isLast should run correcty', () => {
    const {result} = renderHook(() => useReplyComment());
    expect(result.current.isLastInParent(0, 1)).toBeTruthy();
  });
});
