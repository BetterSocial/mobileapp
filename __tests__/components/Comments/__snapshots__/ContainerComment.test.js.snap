// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerComment should run correctly ContainerReply should match snapshot 1`] = `[Function]`;

exports[`ContainerComment should run correctly ReplyComment should match snapshot 1`] = `[Function]`;

exports[`ContainerComment should run correctly Should match snapshot 1`] = `[Function]`;
